File: RestAPI/Models/WinnerRecord.cs
namespace RestAPI.Models
{
    public class WinnerRecord
    {
        public int ID { get; set; }
        public string Name { get; private set; }
        public  TimeSpan Time { get; private set; }
        public DateTime AchievedAt { get; private set; }

        public WinnerRecord(string name, TimeSpan time, DateTime achievedAt)
        {
            Name = name;
            Time = time;
            AchievedAt = achievedAt;
        }
        public WinnerRecord(string name, long time, DateTime achievedAt)
        {
            Name = name;
            Time = TimeSpan.FromMilliseconds(time);
            AchievedAt = achievedAt;
        }
    }
}

File: SortiermeisterWeb/sortiermeister/app/game/lib/api/winnerRecord.ts
import { fetchFromAPI } from "./api";

export async function sendWinnerRecord(data: {
    name: string;
    time: number;
    achievedAt?: string;
}) {
    const payload = {
        name: data.name,
        time: data.time,
        achievedAt: data.achievedAt || new Date().toISOString().split('T')[0],
    };
    console.log(payload);
    await fetchFromAPI("/api/WinnerRecord", {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(payload),
    });
}